src/executable.cpp

```diff
@@ -325,277 +313,134 @@ void Executable::add_symbol(const ExeSymbol &symbol, bool overwrite)
-
-    if (j.find(s_configSection) != j.end())
-    {
-        nlohmann::json &conf = j.at(s_configSection);
-        conf.at("codealign").get_to(m_imageData.codeAlignment);
-        conf.at("dataalign").get_to(m_imageData.dataAlignment);
-        conf.at("codepadding").get_to(m_imageData.codePad);
-        conf.at("datapadding").get_to(m_imageData.dataPad);
-    }
-
-    if (j.find(s_symbolSection) != j.end())
-    {
-        load_symbols(j.at(s_symbolSection), overwrite_symbols);
-    }
-
-    if (j.find(s_sectionsSection) != j.end())
-    {
-        load_sections(j.at(s_sectionsSection));
-    }
-
-    if (j.find(s_objectSection) != j.end())
-    {
-        load_objects(j.at(s_objectSection));
-    }
-
+    nlohmann::json js = nlohmann::json::parse(fs);
+    load_json(js, overwrite_symbols);
     return true;
 }
 
-bool Executable::save_config(const char *file_name) const
+bool Executable::save_config(const char *filename) const
 {
     if (m_verbose)
     {
-        printf("Saving config file '%s'...\n", file_name);
+        printf("Saving config file '%s'...\n", filename);
     }
 
-    nlohmann::json j;
-
-    // Parse the config file if it already exists and update it.
+    nlohmann::json js;
     {
-        std::ifstream fs(file_name);
-
+        std::ifstream fs(filename);
         if (!fs.fail())
         {
-            j = nlohmann::json::parse(fs);
+            js = nlohmann::json::parse(fs);
         }
     }
 
-    if (j.find(s_configSection) == j.end())
-    {
-        j[s_configSection] = nlohmann::json();
-    }
-
-    nlohmann::json &conf = j.at(s_configSection);
-    conf["codealign"] = m_imageData.codeAlignment;
-    conf["dataalign"] = m_imageData.dataAlignment;
-    conf["codepadding"] = m_imageData.codePad;
-    conf["datapadding"] = m_imageData.dataPad;
-
-    // Don't dump if we already have a sections for these.
-    if (j.find(s_symbolSection) == j.end())
-    {
-        j[s_symbolSection] = nlohmann::json();
-        dump_symbols(j.at(s_symbolSection));
-    }
-
-    if (j.find(s_sectionsSection) == j.end())
-    {
-        j[s_sectionsSection] = nlohmann::json();
-        dump_sections(j.at(s_sectionsSection));
-    }
-
-    if (j.find(s_objectSection) == j.end())
-    {
-        j[s_objectSection] = nlohmann::json();
-        dump_objects(j.at(s_objectSection));
-    }
-
-    std::ofstream fs(file_name);
-    fs << std::setw(4) << j << std::endl;
+    save_json(js);
 
+    std::ofstream fs(filename);
+    fs << std::setw(4) << js << std::endl;
     return !fs.fail();
 }
 
-void Executable::load_symbols(nlohmann::json &js, bool overwrite_symbols)
+void Executable::load_json(const nlohmann::json &js, bool overwrite_symbols)
 {
-    if (m_verbose)
-    {
-        printf("Loading external symbols...\n");
-    }
-
-    size_t newSize = m_symbols.size() + js.size();
-    m_symbols.reserve(newSize);
-    m_symbolAddressToIndexMap.reserve(newSize);
-    m_symbolNameToIndexMap.reserve(newSize);
```

Comment:
The reservation for the maps is missing in new code.

Reply:
It's moved to add_symbols, matching what we do with pdb